beater/api/config/agent/handler_test.go

```diff
@@ -342,6 +342,26 @@ func TestIfNoneMatch(t *testing.T) {
 	assert.Equal(t, "123", ifNoneMatch(fromQueryArg("123")))
 }
 
+func TestAgentConfigTraceContext(t *testing.T) {
+	kibanaCfg := libkibana.DefaultClientConfig()
+	kibanaCfg.Host = "testKibana:12345"
+	client := kibana.NewConnectingClient(&kibanaCfg)
+	handler := Handler(client, &config.AgentConfig{Cache: &config.Cache{Expiration: 5 * time.Minute}})
+	_, spans, _ := apmtest.WithTransaction(func(ctx context.Context) {
+		// When the handler is called with a context containing
+		// a transaction, the underlying Kibana query should create a span
+		w := httptest.NewRecorder()
+		r := httptest.NewRequest(http.MethodPost, "/backend", convert.ToReader(m{
+			"service": m{"name": "opbeans"}}))
+		r = r.WithContext(ctx)
+		c := request.NewContext()
+		c.Reset(w, r)
+		handler(c)
+	})
+	require.Len(t, spans, 1)
```

Comment:
shouldn't there be another span from the apmhttp instrumentation now?

Reply:
No because the first call is not successful.